src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -3680,6 +3681,311 @@ public sealed class C
+    public partial class C : ISNode
+    {
+        public static IEqualityComparer<C> ValueComparer => CEqualityComparer.Instance;
+
+        public bool ValueEquals(C other) => ValueComparer.Equals(this, other);
+        public int ValueGetHashCode() => ValueComparer.GetHashCode(this);
+
+        /// <summary>
+        /// Gets a value indicating the type of object implementing <see cref=""ISNode"" />.
+        /// </summary>
+        public virtual SNodeKind SNodeKind
+        {
+            get
+            {
+                return SNodeKind.C;
+            }
+        }
+
+        [DataMember(Name = ""prop"", IsRequired = false, EmitDefaultValue = false)]
+        public virtual string Prop { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref=""C"" /> class.
+        /// </summary>
+        public C()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref=""C"" /> class from the supplied values.
+        /// </summary>
+        /// <param name=""prop"">
+        /// An initialization value for the <see cref=""P:Prop"" /> property.
+        /// </param>
+        public C(string prop)
+        {
+            Init(prop);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref=""C"" /> class from the specified instance.
+        /// </summary>
+        /// <param name=""other"">
+        /// The instance from which the new instance is to be initialized.
+        /// </param>
+        /// <exception cref=""ArgumentNullException"">
+        /// Thrown if <paramref name=""other"" /> is null.
+        /// </exception>
+        public C(C other)
+        {
+            if (other == null)
+            {
+                throw new ArgumentNullException(nameof(other));
+            }
+
+            Init(other.Prop);
+        }
+
+        ISNode ISNode.DeepClone()
+        {
+            return DeepCloneCore();
+        }
+
+        /// <summary>
+        /// Creates a deep copy of this instance.
+        /// </summary>
+        public virtual C DeepClone()
+        {
+            return (C)DeepCloneCore();
+        }
+
+        private ISNode DeepCloneCore()
+        {
+            return new C(this);
+        }
+
+        private void Init(string prop)
+        {
+            Prop = prop;
+        }
+    }
+}";
+
+            const string ExpectedComparerClass =
+@"using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+
+namespace N
+{
+    /// <summary>
+    /// Defines methods to support the comparison of objects of type C for equality.
+    /// </summary>
+    [GeneratedCode(""Microsoft.Json.Schema.ToDotNet"", """ + VersionConstants.FileVersion + @""")]
+    internal sealed class CEqualityComparer : IEqualityComparer<C>
+    {
+        internal static readonly CEqualityComparer Instance = new CEqualityComparer();
+
+        public bool Equals(C left, C right)
```

Comment:
>bool [](start = 15, length = 4)

Equality comparer methods should not be virtual. Derived class can make its own and then delegate to the base class.